code/modules/clothing/chameleon.dm

```diff
@@ -149,6 +153,18 @@
 	select_look(owner)
 	return 1
 
+/datum/action/item_action/chameleon/change/proc/emp_randomise()
+	START_PROCESSING(SSprocessing, src)
```

Comment:
Don't use SSprocessing, it just means you're trying to update basically every tick.

Use a timer set to go off every 3 seconds